# IX. Model Inference

In [17]:
import pandas as pd
import numpy as np
import sklearn

In [19]:
sklearn.__version__

'1.2.1'

### Load Model

In [2]:
import joblib

with open('model_linreg.pkl', 'rb') as file_1:
  model_linreg = joblib.load(file_1)

with open('model_standardscaler.pkl', 'rb') as file_2:
  model_standardscaler = joblib.load(file_2)

with open('model_onehotencoder.pkl', 'rb') as file_3:
  model_onehotencoder = joblib.load(file_3)

with open('model_ordinalencoder.pkl','rb') as file_4:
  model_ordinalencoder = joblib.load(file_4)

In [3]:
#Membuat data baru
df_inf = pd.DataFrame({
    'hour':[9,17,22,13,19],
    'name':['UberX','Lux Black XL','UberPool','Lyft','Black'],
    'distance':[2.53,3.14,1.50,13.22,15.99],
    'surge_multiplier':[1.0,2.50,1.75,1.0,2.0]
})
df_inf

hour          name  distance  surge_multiplier
0     9         UberX      2.53              1.00
1    17  Lux Black XL      3.14              2.50
2    22      UberPool      1.50              1.75
3    13          Lyft     13.22              1.00
4    19         Black     15.99              2.00

#### Encode

##### One Hot Encoder

In [4]:
df_inf_ohe = model_onehotencoder.transform(df_inf[['name']])

##### Ordinal Encoder

In [5]:
df_inf_ordenc = model_ordinalencoder.transform(df_inf[['surge_multiplier']])

#### Scaling

In [6]:
df_inf_scaler = model_standardscaler.transform(df_inf[['hour','distance']])

In [7]:
df_inf_final = np.concatenate([df_inf_ohe,df_inf_ordenc,df_inf_scaler],axis=1) #digabungkan
df_inf_final

array([[ 0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  0.        ,  1.        ,
         0.        ,  0.        ,  0.        , -0.37587233,  0.3637116 ],
       [ 0.        ,  0.        ,  0.        ,  0.        ,  1.        ,
         0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  5.        ,  0.77554708,  0.94046538],
       [ 0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  1.        ,  0.        ,
         0.        ,  0.        ,  3.        ,  1.49518421, -0.61015133],
       [ 0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         1.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        ,  0.19983738, 10.47108514],
       [ 1.        ,  0.        ,  0.        ,  0.        ,  0.        ,
         0.        ,  0.        ,  0.        , 

In [8]:
#hasil prediksi
df_inf_predict = model_linreg.predict(df_inf_final)
df_inf_predict

array([10.84528637, 57.54459095, 20.83759308, 40.42962933, 78.42238235])

In [13]:
prediksi = pd.DataFrame(df_inf_predict,columns=['Hasil Prediksi'])
prediksi

Hasil Prediksi
0       10.845286
1       57.544591
2       20.837593
3       40.429629
4       78.422382

In [16]:
prediksi_final = pd.concat([df_inf,prediksi],axis=1)
prediksi_final

hour          name  distance  surge_multiplier  Hasil Prediksi
0     9         UberX      2.53              1.00       10.845286
1    17  Lux Black XL      3.14              2.50       57.544591
2    22      UberPool      1.50              1.75       20.837593
3    13          Lyft     13.22              1.00       40.429629
4    19         Black     15.99              2.00       78.422382

Interpretasi: 

Hasil prediksi apabila ada orang yang memesan UberX di jam 9 pagi dengan jarak tempuh 2.53 km dan surge multiplier 1.00 adalah 10.8 dollar, dan seterusnya